File: 2582-pass-the-pillow/2582-pass-the-pillow.py
class Solution:
    def passThePillow(self, n: int, time: int) -> int:
        
        p = 1
        dirc_forward = True
        
        while time > 0:
            if dirc_forward:
                p += 1
                if p == n:
                    dirc_forward = False
            else:
                p -= 1
                if p == 1:
                    dirc_forward = True
                    
            time -= 1
        
        return p
            

File: 2582-pass-the-pillow/2582-pass-the-pillow.cpp
class Solution {
public:
    int passThePillow(int n, int time) {
       int t2 = time %( 2*(n-1));
       if(t2 > (n-1)){
           int t3 = t2 - (n-1);
           return n-t3;
       } 
       return t2 + 1;
    }
};